### ISSUE TITLE
Improve chunk upload protocol with websocket

### ISSUE THREAD

**Issue by tmm360** [ethersphere/bee#4854]
#### Summary
Current websocket protocol is heavily suboptimized. A better protocol to improve upload speed is possible and necessary.

#### Motivation
Current protocol send one chunk at time, and waits ack from server.
All the time the network is waiting for something, or to receive new chunk data, or confirmation that data has been received. The network line is always empty of data.

Instead, to have a continuous stream of chunks is a really better solution.

#### Implementation
I've implemented an alternative protocol into an experimental reverse proxy project that I've called "BeeTurbo". It is putted directly in front of bee.

- server implementation: https://github.com/Etherna/bee-turbo/blob/952393dea4fd0c2991c5a693e7bd51d089abe182/src/BeeTurbo/Tools/ChunkStreamTurboProcessor.cs
- client implementation: https://github.com/Etherna/etherna-sdk-dotnet/blob/15270e7a4aa4e3cc06d34aadaa12f9333522bada/src/EthernaSdk.Users.Gateway/Tools/ChunkWebSocketTurboUploader.cs

Basically, it receives a stream of chunks, where each chunk data is prefixed by its byte size (uint16). It ends when server receives from client indication that a message is the last one.

With this experimental solution that uses a reverse proxy to receive a stream, and convert it to bee's current ws protocol locally, I've achieved to improve the bandwidth of chunks upload on prod from ~1Mb/s to ~30Mb/s.

#### Drawbacks
The new api is not retro-compatible, an alternative endpoint have to be introduced

**Comment by tmm360** [comment]
Closing because I've understood that this solution has several limits.
